src/client/dfs/dfs.c

```diff
@@ -3277,7 +3277,7 @@ dfs_obj_local2global(dfs_t *dfs, dfs_obj_t *obj, d_iov_t *glob)
 	oid_cp(&obj_glob->parent_oid, obj->parent_oid);
 	uuid_copy(obj_glob->coh_uuid, coh_uuid);
 	uuid_copy(obj_glob->cont_uuid, cont_uuid);
-	strncpy(obj_glob->name, obj->name, DFS_MAX_NAME + 1);
+	snprintf(obj_glob->name, DFS_MAX_NAME + 1, "%s", obj->name);
```

Comment:
In the past, I've addressed this one by simply putting this after strncpy

obj_glob->name[DFX_MAX_NAME] = 0;

If this is performance sensitive at all, this would be more performant.

Reply:
ok i updated this

Reply:
Just going through this same thing with dfuse and I think the strncpy() should be using DFS_NAME_MAX rather than DFS_NAME_MAX + 1, with the '\0' assignment only need if target initial memory is uninitialised.

Just going through this with dfuse and I think this code would copy up to 256 bytes, then truncate it, although the source should only have 255 bytes max anyway so the end result is the same. dfs_obj_global2local() looks to have the same code as was similar to this function before the PR but wasn't picked up by coverity so there is more we can do however.

https://github.com/daos-stack/daos/pull/6825 is my current change to dfuse, I could update that to include changes to dfs is required.